enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,179 @@
+
+### Goals
+
+* Allow deployment in networks where multicast VRRP traffic is forbidden or non-functional.
+* No loss of functionality from the current multicast implementation.
+* Prevent VRRP ID collisions during deployment.
+
+### Non-Goals
+
+* A switchable implementation. It does not need to be an installer choice.
+* Allow masters to be deployed on different subnets (not on the same L2 segment).
+
+## Proposal
+
+Change the way we configure and update Keepalived to manage the VIPs using unicast peer lists.
+
+### User Stories
+
+#### Story 1 - Allow deployment where networking doesn't permit multicast
+
+As an administrator, I want to deploy OpenShift on bare metal nodes with 
+installer-provisioned infrastructure (IPI) where the networking doesn't permit multicast.
+Some customers have restrictions on multicast network traffic (it impacts telco/edge customers since it’s not possible to use multicast between the regional and far-edge locations).
+The current deployment supports high availability for API and default ingress using multicast traffic for VRRP management of the VIP addresses.
+
+
+#### Story 2 - Allow multiple deployments on the same network
+
+For PoC and test deployments we might need to run multiple clusters on the same network (requiring a dedicated network is a heavy dependency in non-virtualized environments), while we’d still recommend isolated networks for production it would be desirable to enable deployment of multiple clusters on a single broadcast domain.
+Keepalived uses the `virtual_router_id` field to specify to which VRRP router the [instance](https://www.keepalived.org/doc/configuration_synopsis.html#vrrp-instance-definitions-synopsis) belongs,`virtual_router_id` range is 1 - 255.
+Each baremetal deployment includes two virtual routers:
+* API
+* Ingress
+
+Although the `virtual_router_id` values for both API and Ingress are calculated using [checksum function based on cluster name](https://github.com/openshift/baremetal-runtimecfg/blob/master/pkg/config/node.go#L162-#L171) we can still end up in a situation where virtual routers from different deployments or services share the same `virtual_router_id`.
+
+### Implementation Details/Notes/Constraints
+
+Unlike multicast mode, in unicast we should specify and keep up to date all the group members in 'unicast_peer' section at [Keepalived conf](https://www.keepalived.org/manpage.html) for each VIP. Lets first present briefly the supported VIPs and the group members for each one of them.
+
+As mentioned above, `api-vip` and `ingress-vip` are supported to enable high availability.
+
+The `api-vip` should resides on the bootstrap instance during the bootstrap phase and move to one of the control plane nodes after the bootstrap process has completed and the bootstrap instance is stopped.
+The group members for the `api-vip` should include the bootstrap node and the control plane nodes.
```

Comment:
> and the bootstrap instance is stopped.

hmm.. this seems odd, why would stopping the bootstrap host be required, why can we move as soon as the control-plane nodes have capacity to host the VIP?

Reply:
My bad, the VIP moves to one of the control plane nodes as soon as he can host it.
I'll update that .